pyproject.toml

```diff
@@ -60,20 +61,20 @@ doc = [
     "meshzoo>=0.7",
     "numpydoc>=1.1",
     "pykrige>=1.5,<2",
-    "pyvista>=0.29",
-    "sphinx>=4",
+    "pyvista>=0.40",
+    "sphinx>=7",
     "sphinx-gallery>=0.8",
-    "sphinx-rtd-theme>=1,<1.1",
+    "sphinx-rtd-theme>=2",
     "sphinxcontrib-youtube>=1.1",
 ]
 plotting = [
     "matplotlib>=3",
-    "pyvista>=0.29",
+    "pyvista>=0.40",
 ]
 rust = ["gstools_core>=0.2.0,<1"]
 test = ["pytest-cov>=3"]
 lint = [
-    "black",
+    "black>=23,<24",
```

Comment:
black 24 introduced a blank line under the file doc-string. Without the cap, I would update almost all files to add a line there. And since black updates there policy changes once a year, this cap is reasonable.

Reply:
yes, I guess. So you would prefer configuring this instead of pinning the black version?

Reply:
It seems, this is not configurable since it was standardized with black 24: https://github.com/psf/black/issues/4042

We could update to 24 (add blank line) and add a preview job:
```
python -m black --preview --diff --color .
```
That will show diffs from upcoming conventions but wont raise an error.

Reply:
Damn conformists ;-)

I mean, we don't forward the dependency to the user, so if this really is so complicated, I say do what's easiest for you.

Reply:
Applied black 24 and added a preview job.